analyses/cell-type-neuroblastoma-04/exploratory-notebooks/scanvi-singler-comparison.Rmd

```diff
@@ -0,0 +1,391 @@
+
+```{r}
+celltype_df <- scuttle::makePerCellDF(
+  merged_sce,
+  use.coldata = c("sample_id", "consensus_celltype_annotation"),
+  use.dimred = c("UMAP")
+) |>
+  # there ARE repeated barcodes so we need to keep cell_id around
+  tibble::rownames_to_column("cell_id") |>
+  dplyr::rename(
+    UMAP1 = UMAP.1,
+    UMAP2 = UMAP.2,
+    consensus_annotation = consensus_celltype_annotation
+  ) |>
+  dplyr::left_join(validation_df, by = "consensus_annotation") |>
+  # recode consensus NAs to "unknown_consensus"
+  dplyr::mutate(
+    validation_group_annotation = ifelse(
+      is.na(validation_group_annotation), "unknown_consensus", validation_group_annotation
+    )
+  ) |>
+  dplyr::select(-consensus_annotation) |>
+  # merge in the annotations
+  dplyr::left_join(singler_df, by = "cell_id") |>
+  dplyr::left_join(scanvi_df, by = "cell_id") |>
+  # make it tidy
+  tidyr::pivot_longer(
+    c(validation_group_annotation, singler, scanvi), 
+    names_to = "celltype_method", 
+    values_to = "label"
+  ) |>
+  harmonize_celltypes(label, label_recoded) 
+```
+
+
+## UMAP
+
+This section displays the UMAP for the _merged object_ (not integrated) with cell type labels harmonized for comparison.
+We show a panel for each cell type method.
+
+```{r, fig.width = 12, fig.height = 8}
+ggplot(celltype_df) +
+  aes(x = UMAP1, y = UMAP2, color = label_recoded) +
+  geom_point(alpha = 0.5, size = 0.2) +
+  scale_color_manual(values = recoded_celltype_pal) +
+  facet_wrap(vars(celltype_method)) +
+  coord_equal() +
+  theme(
+    legend.position = "bottom", 
+    axis.text = element_blank(), 
+    axis.ticks = element_blank()
+  ) +
+  guides(color = guide_legend(override.aes = list(size = 2, alpha = 1)))
+```
+
+
+## Heatmap
+
+This section shows a stacked heatmap comparing the `SingleR` and `scANVI` annotations to the consensus validation groups.
+
+```{r, fig.height = 13, fig.width = 12} 
+# make a wide version for the heatmap
+celltype_wide_df <- celltype_df |>
+  dplyr::select(-label) |>
+  tidyr::pivot_wider(
+    names_from = celltype_method,
+    values_from = label_recoded
+  )
+
+list(
+  "SingleR annotations" = make_jaccard_matrix(celltype_wide_df, "validation_group_annotation", "singler"),
+  "scANVI annotations" = make_jaccard_matrix(celltype_wide_df, "validation_group_annotation", "scanvi")
+) |>
+  purrr::imap(
+    \(jaccard_mat, name) create_single_heatmap(
+        mat = jaccard_mat,
+        row_title = name,
+        column_title = "Consensus validation groups",
+        keep_legend_name = "SingleR annotations" # arbitrary
+    )
+  ) |>
+  # concatenate vertically into HeatmapList object
+  purrr::reduce(ComplexHeatmap::`%v%`) |>
+  ComplexHeatmap::draw(heatmap_legend_side = "right")
+```
+
+
+We'll also show some statistics from a confusion matrix comparison between `SingleR` and `scANVI`:
+
+```{r}
+# ensure factors are compatible for caret
```

Comment:
I want to confirm that this is looking at agreement between all labels in the immune zoom reference?